prepare-build.sh

```diff
@@ -47,10 +47,16 @@ function pull_dir {
         [ "$1" != "." ]   && pushd $1 > /dev/null
         git symbolic-ref HEAD &> /dev/null
         if [ $? -eq 0 ] ; then
-            echo -e "\e[32mPulling $1\e[39m"
-            git pull --rebase
+            if git remote get-url upstream > /dev/null
+            then
+                echo -e "\e[32mPulling $1\e[39m"
+                git pull upstream --rebase "$2"
+            else
+                echo -e "\e[32mPulling $1\e[39m"
```

Comment:
Could you extract the echo out of the conditions to avoid the duplicated code ? :)

Reply:
Yes, done.